src/server/index.jsx

```diff
@@ -47,18 +51,34 @@ server
   .get('/status', (req, res) => {
     res.sendStatus(200);
   })
-  .get('/*', async (req, res) => {
+  .get('/*', async ({ url }, res) => {
     try {
-      const html = await render({
-        req,
-        res,
-        routes,
-        document: Document,
-        assets,
-      });
-      res.send(html);
-    } catch (error) {
-      res.json(error);
+      const sheet = new ServerStyleSheet();
+      const data = await loadInitialData(url, routes);
+
+      const app = renderToString(
+        sheet.collectStyles(
+          <ServerUni url={url} routes={routes} data={data} />,
+        ),
+      );
+
+      const helmet = Helmet.renderStatic();
+
+      const styleTags = sheet.getStyleElement();
+
+      const doc = renderToStaticMarkup(
+        <Document
+          assets={assets}
+          app={app}
+          data={data}
+          styleTags={styleTags}
+          helmet={helmet}
+        />,
+      );
+
+      res.send(`<!doctype html>${doc}`);
+    } catch ({ message }) {
+      res.status(404).send(message);
```

Comment:
We would want to handle this with more nuance, but I did not want to change the underlying implementation too much (apart from return a 404 on the catch-all route).